source/PlayerInfo.cpp

```diff
@@ -1522,24 +1522,9 @@ bool PlayerInfo::TakeOff(UI *ui)
 			}
 			else
 				ship->Recharge(hasSpaceport);
-
-			if(ship != flagship)
-			{
-				ship->Cargo().SetBunks(ship->Attributes().Get("bunks") - ship->RequiredCrew());
-				cargo.TransferAll(ship->Cargo());
-			}
-			else
-			{
-				// Your flagship takes first priority for passengers but last for cargo.
-				desiredCrew = ship->Crew();
-				ship->Cargo().SetBunks(ship->Attributes().Get("bunks") - desiredCrew);
-				for(const auto &it : cargo.PassengerList())
-					cargo.TransferPassengers(it.first, it.second, ship->Cargo());
-			}
 		}
-	// Load up your flagship last, so that it will have space free for any
-	// plunder that you happen to acquire.
-	cargo.TransferAll(flagship->Cargo());
+
+	DistributeCargo();
```

Comment:
Indeed, but it needs to be called in both places, as TakeOffIfReady() can call TakeOff() without first calling CheckWarningsAndTakeOff().  Perhaps have Player::TakeOff() check to see if DistributeCargo() was already called?

Reply:
The second call to DistributeCargo isn't that bad, as most cargo has already been loaded.  However, if you want to prevent it, you can have PlanetPanel::TakeOff take a distributeCargo bool and pass it on to player.TakeOff().